Modules/_winapi.c

```diff
@@ -774,23 +774,111 @@ gethandle(PyObject* obj, const char* name)
     return ret;
 }
 
+static PyObject* sortenvironmentkey(PyObject *module, PyObject *item) {
+    return _winapi_LCMapStringEx_impl(NULL, LOCALE_NAME_INVARIANT, LCMAP_UPPERCASE, item);
+}
+
+static PyMethodDef sortenvironmentkey_def = {
+    "sortenvironmentkey",
+    _PyCFunction_CAST(sortenvironmentkey),
+    METH_O,
+    ""
+};
+
+static PyObject *
+normalize_environment(PyObject* environment) {
+    PyObject *result, *keys, *keyfunc, *sort, *args, *kwargs;
+
+    keys = PyMapping_Keys(environment);
+    if (keys == NULL) {
+        return NULL;
+    }
+
+    keyfunc = PyCFunction_New(&sortenvironmentkey_def, NULL);
+    sort = PyObject_GetAttrString(keys, "sort");
+    args = PyTuple_New(0);
+    kwargs = PyDict_New();
+    PyDict_SetItemString(kwargs, "key", keyfunc);
+    if (PyObject_Call(sort, args, kwargs) == NULL) {
+        goto error;
+    }
+
+    result = PyDict_New();
+
+    for (int i=0; i<PyList_GET_SIZE(keys); i++) {
```

Comment:
Iteration over the dict items can be done more efficiently with [PyDict_Next](https://docs.python.org/3/c-api/dict.html#c.PyDict_Next), but I am not sure `environment` is guaranteed to be a dict

Reply:
The `environment` is passed from https://github.com/aisk/cpython/blob/getenvironment/Modules/_winapi.c#L1137 , so it can be any mapping object, so I think we should keep this.